diff -Naur src-orig/menu.h src-new/menu.h
--- a/menu.h
+++ b/menu.h
@@ -3,7 +3,7 @@
 
 #include <stdbool.h>
 #include <stdint.h>
-#include <systemd/sd-bus.h>
+#include "sd-bus.h"
 
 typedef struct MangobarMenu MangobarMenu;
 typedef struct MangobarMenuNode MangobarMenuNode;
diff -Naur src-orig/meson.build src-new/meson.build
--- a/meson.build
+++ b/meson.build
@@ -6,6 +6,10 @@
 cc = meson.get_compiler('c')
 add_project_arguments(cc.get_supported_arguments('-Wno-unused-parameter'),
                       language: 'c')
+
+sdbus_dep = dependency('libsystemd', 'libelogind', 'basu')
+add_project_arguments('-D' + 'SDBUS_' + sdbus_dep.name().to_upper(), language: 'c')
+
 deps = [
   dependency('threads'),
   dependency('wayland-client'),
@@ -16,7 +20,7 @@
   dependency('pango'),
   dependency('pangocairo'),
   dependency('libpulse'),
-  dependency('libsystemd'),
+  sdbus_dep,
   dependency('libudev'),
   dependency('gdk-pixbuf-2.0'),
   dependency('alsa'),
diff -Naur src-orig/sd-bus.h src-new/sd-bus.h
--- /dev/null
+++ b/sd-bus.h
@@ -0,0 +1,17 @@
+#ifndef MANGOBAR_SD_BUS_H
+#define MANGOBAR_SD_BUS_H
+
+#if defined(SDBUS_LIBSYSTEMD)
+#include <systemd/sd-bus.h>
+#include <systemd/sd-bus-vtable.h>
+#elif defined(SDBUS_LIBELOGIND)
+#include <elogind/sd-bus.h>
+#include <elogind/sd-bus-vtable.h>
+#elif defined(SDBUS_BASU)
+#include <basu/sd-bus.h>
+#include <basu/sd-bus-vtable.h>
+#else
+#error "No supported sd-bus implementation selected"
+#endif
+
+#endif
diff -Naur src-orig/tray.c src-new/tray.c
--- a/tray.c
+++ b/tray.c
@@ -12,8 +12,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <systemd/sd-bus.h>
-#include <systemd/sd-bus-vtable.h>
+#include "sd-bus.h"
 #include <unistd.h>
 
 #define WATCHER_PATH "/StatusNotifierWatcher"
